res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -2253,358 +1888,222 @@ NumarkMixtrack3.PFLButton = function(channel, control, value, status, group) {
-    var decknum = group.substring(26, 27);
-    decknum = NumarkMixtrack3.deckFromGroup("[Channel" + decknum + "]");
-    var deck = NumarkMixtrack3.decks["D" + decknum];
-
-    if (!deck.shiftKey && !deck.PADMode && !deck.TapDown) {
-        parameterSoftTakeOver("[QuickEffectRack1_[Channel" + decknum + "]]", "super1", value);
-    }
-
-    if (FXmode === 1) {
-        if (deck.shiftKey) {
-            parameterSoftTakeOver("[EffectRack1_EffectUnit" + decknum +
-                "_Effect1]", "parameter4", value);
-        }
-        if (deck.PADMode) {
-            parameterSoftTakeOver("[EffectRack1_EffectUnit" + decknum +
-                "_Effect2]", "parameter4", value);
-        }
-        if (deck.TapDown) {
-            parameterSoftTakeOver("[EffectRack1_EffectUnit" + decknum +
-                "_Effect3]", "parameter4", value);
-        }
+    var deck = NumarkMixtrack3.deckFromGroup("[Channel" + group.substring(26, 27) + "]");
+    var decknum = deck.decknum;
+    var focusedEffect = deck.getFocusedEffect();
+
+    // default behavior is to control filter
+    // when shifted, change parameters of focused effect
+    if (deck.shiftKey && focusedEffect) {
+        parameterSoftTakeOver(
+            "[EffectRack1_EffectUnit" + decknum + "_Effect" + focusedEffect + "]", "parameter4", value
+        );
     } else {
-        if (deck.shiftKey) {
-            parameterSoftTakeOver("[EffectRack1_EffectUnit1_Effect1]", "parameter4", value);
-        }
-        if (deck.PADMode) {
-            parameterSoftTakeOver("[EffectRack1_EffectUnit2_Effect1]", "parameter4", value);
-        }
-        if (deck.TapDown) {
-            parameterSoftTakeOver("[EffectRack1_EffectUnit3_Effect1]", "parameter4", value);
-        }
+        parameterSoftTakeOver("[QuickEffectRack1_[Channel" + decknum + "]]", "super1", value);
     }
 };
 
 NumarkMixtrack3.loop_in = function(channel, control, value, status, group) {
-    var decknum = NumarkMixtrack3.deckFromGroup(group);
+    var deck = NumarkMixtrack3.deckFromGroup(group);
 
     if (value === DOWN) {
-        engine.setValue("[Channel" + decknum + "]", "loop_in", value ? 1 : 0);
+        engine.setValue(deck.group, "loop_in", value ? 1 : 0);
     }
 };
 
 NumarkMixtrack3.loop_out = function(channel, control, value, status, group) {
-    var decknum = NumarkMixtrack3.deckFromGroup(group);
+    var deck = NumarkMixtrack3.deckFromGroup(group);
 
     if (value === DOWN) {
-        engine.setValue("[Channel" + decknum + "]", "loop_out", value ? 1 : 0);
+        engine.setValue(deck.group, "loop_out", value ? 1 : 0);
     }
 };
 
 NumarkMixtrack3.reloop_exit = function(channel, control, value, status, group) {
-    var decknum = NumarkMixtrack3.deckFromGroup(group);
+    var deck = NumarkMixtrack3.deckFromGroup(group);
 
     if (value === DOWN) {
-        engine.setValue("[Channel" + decknum + "]", "reloop_exit", value ? 1 : 0);
+        engine.setValue(deck.group, "reloop_exit", value ? 1 : 0);
```

Comment:
You won't necessarily have to change this because the old reloop_exit name will be aliased, but #1187 will change this confusing name to reloop_toggle.